extensions/ql-vscode/src/remote-queries/sarif-processing.ts

```diff
@@ -19,77 +20,63 @@ export function extractAnalysisAlerts(
   alerts: AnalysisAlert[],
   errors: string[]
 } {
-  if (!sarifLog) {
-    return { alerts: [], errors: ['No SARIF log was found'] };
-  }
-
-  if (!sarifLog.runs) {
-    return { alerts: [], errors: ['No runs found in the SARIF file'] };
-  }
-
-  const errors: string[] = [];
   const alerts: AnalysisAlert[] = [];
+  const errors: string[] = [];
 
-  for (const run of sarifLog.runs) {
-    if (!run.results) {
-      errors.push('No results found in the SARIF run');
-      continue;
-    }
-
-    for (const result of run.results) {
-      if (!result.message?.text) {
-        errors.push('No message found in the SARIF result');
-        continue;
-      }
-
-      const { message, errors: messageErrors } = extractMessage(result.message?.text, result);
-      if (messageErrors.length > 0) {
-        errors.push(...messageErrors);
-        continue;
-      }
-
-      const rule = tryGetRule(run, result);
-      const severity = tryGetSeverity(run, result, rule) || defaultSeverity;
-
-      const { codeFlows, errors: codeFlowsErrors } = extractCodeFlows(result);
-      if (codeFlowsErrors.length > 0) {
-        errors.push(...codeFlowsErrors);
-        continue;
-      }
-
-      if (!result.locations) {
-        errors.push('No locations found in the SARIF result');
-        continue;
-      }
-
-      const shortDescription = getShortDescription(rule, message!);
-
-      for (const location of result.locations) {
-        const { processedLocation, errors: locationErrors } = extractLocation(location);
-
-        if (locationErrors.length > 0) {
-          errors.push(...locationErrors);
+  try {
+    for (const run of sarifLog.runs!) {
+      for (const result of run.results!) {
+        try {
+          alerts.push(...extractResultAlerts(run, result));
+        } catch (e) {
+          errors.push(`Error when processing SARIF result: ${e}`);
           continue;
         }
-
-        const analysisAlert: AnalysisAlert = {
-          message: message!,
-          shortDescription,
-          filePath: processedLocation!.filePath,
-          severity,
-          codeSnippet: processedLocation!.codeSnippet,
-          highlightedRegion: processedLocation!.highlightedRegion,
-          codeFlows: codeFlows
-        };
-
-        alerts.push(analysisAlert);
       }
     }
+  } catch (e) {
+    errors.push(`Error when processing SARIF run: ${e}`);
   }
 
   return { alerts, errors };
 }
 
+function extractResultAlerts(
+  run: sarif.Run,
+  result: sarif.Result
+): AnalysisAlert[] {
+  const alerts: AnalysisAlert[] = [];
+
+  const message = getMessage(result);
+  const rule = tryGetRule(run, result);
+  const severity = tryGetSeverity(run, result, rule) || defaultSeverity;
+  const codeFlows = getCodeFlows(result);
+  const shortDescription = getShortDescription(rule, message!);
+
+  for (const location of result.locations!) {
```

Comment:
If a result has no location associated with it (this can happen with a query that doesn't include locations), does this array exist? I'm not sure. Maybe safer to just do:

```suggestion
  for (const location of result.locations ?? []) {
```